easybuild/easyconfigs/c/CP2K/CP2K-5.1-foss-2020a.eb

```diff
@@ -0,0 +1,45 @@
+name = 'CP2K'
+version = '5.1'
+
+homepage = 'https://www.cp2k.org/'
+description = """CP2K is a freely available (GPL) program, written in Fortran 95, to perform atomistic and molecular
+ simulations of solid state, liquid, molecular and biological systems. It provides a general framework for different
+ methods such as e.g. density functional theory (DFT) using a mixed Gaussian and plane waves approach (GPW), and
+ classical pair and many-body potentials. """
+
+toolchain = {'name': 'foss', 'version': '2020a'}
+toolchainopts = {'pic': True}
+
+source_urls = ['https://github.com/cp2k/cp2k/releases/download/v%(version)s.0/']
+sources = [SOURCELOWER_TAR_BZ2]
+patches = [
+    'CP2K-2.4.0-fix_compile_date_lastsvn.patch',
+]
+checksums = [
+    'e23613b593354fa82e0b8410e17d94c607a0b8c6d9b5d843528403ab09904412',  # cp2k-5.1.tar.bz2
+    '02475cbe24c8d4ba27037c826adf8a534cad634c3c4e02c21d743f5284516bda',  # CP2K-2.4.0-fix_compile_date_lastsvn.patch
+]
+
+dependencies = [
+    ('Libint', '1.1.6'),
+    ('libxc', '3.0.1'),
+    ('libxsmm', '1.16.1'),
+    ('FFTW', '3.3.8'),
+    ('PLUMED', '2.6.0', '-Python-3.8.2'),
+]
+
+builddependencies = [
+    ('flex', '2.6.4'),
+    ('Bison', '3.5.3'),
+]
+
+# values to use here are +1 those used for building Libint
+# see https://github.com/cp2k/cp2k/blob/master/cp2k/tools/hfx_tools/libint_tools/README_LIBINT
+extradflags = "-D__LIBINT_MAX_AM=6 -D__LIBDERIV_MAX_AM1=5"
+
+# regression test reports failures
+ignore_regtest_fails = True
+
+modextravars = {'OMP_NUM_THREADS': '1'}
```

Comment:
I agree it's a good idea to set this, but other sites may not want it here, so better remove it.

Reply:
I cannot really think a use case when `OMP_NUM_THREADS` is not set by "hand" or it has a meaningful value other than 1

Reply:
Setting OMP_NUM_THREADS in a userlevel code like this is bad.
Especially when doing it without checking if it is already set.
I strongly oppose having this set in CP2K!!

Reply:
This is the POPT version, so it is MPI only. In case of PSMP which is OMP+MPI hybrid yes, it makes sense not to set OMP_NUM_THREADS. Anyway, I give it up and I will not set, but do not be surprised when users will ask why the performance of this version is terribly bad. (because it will be terribly bad, if user runs it N cpus, N^2 threads will be started)

Reply:
Hmm, I thought cp2k had been fixed to build all versions at once as it should do.
In that case it's less bad.